source/postprocess/composition_velocity_statistics.cc

```diff
@@ -177,11 +185,25 @@ namespace aspect
         {
           selected_fields = Utilities::split_string_list(prm.get("Names of selected compositional fields"));
 
-          AssertThrow((selected_fields.size() > 0) &&
-                      (selected_fields.size() <= this->n_compositional_fields()),
-                      ExcMessage("The length of the list of names for the compositional "
-                                 "fields for which the RMS velocity is to be summed must be larger than zero "
-                                 "and smaller or equal to the number of compositional fields."));
+          // Check that the names given as selected_fields are actually fields.
+          for (const std::string &field_name: selected_fields)
+            {
+              std::vector<std::string> field_names = this->get_parameters().names_of_compositional_fields;
+              AssertThrow(std::find(field_names.begin(),
+                                    field_names.end(),
+                                    field_name)
+                          != field_names.end(),
```

Comment:
Could you simplify this:

```suggestion
              AssertThrow(this->introspection().compositional_name_exists(field_name),
```